### PR TITLE
Onboarding Phase 2 / Step 1

### PR DESCRIPTION
**Pull request by kornelijepetak**
Step 1 of #1647 

For some reason, the unicode emoji does not break the line properly and gets into a new line even though there's enough space. But I wasn't able to find a good way to fix this 😢

So, it looks like this:

image

@pe0ny [the designs](https://zpl.io/b6lMM6j) show the tooltip with two lines, but is the `Tap to edit` in the new line only because it was too long, or should it always be in the new line?

🦑 whenever.

### CODE REVIEW

**Comment by kornelijepetak** [comment]
Ref to Step 2: #2575

**Comment by daividssilverio** [comment]
Have you tried adjusting the minWidth of the `TooltipWithLeftTopArrow.axml` to make the emoji fit on the first line?
If that doesn't work, and we really want that emoji on the first line, using multiple TextViews could do the job.

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 1]
```diff
@@ -64,6 +71,24 @@ protected override void OnCreate(Bundle bundle)
             disposeBag.Add(ViewModel.WeakSubscribe<PropertyChangedEventArgs>(nameof(ViewModel.SyncingProgress), onSyncChanged));
 
             setupStartTimeEntryOnboardingStep();
+
+            mainRecyclerView = FindViewById<MainRecyclerView>(Resource.Id.MainRecyclerView);
+            mainRecyclerView.FirstTimeEntryView
+                            .ObserveOn(SynchronizationContext.Current)
+                            .Subscribe(updateView)
```
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply Toggl.Giskard/Activities/MainActivity.cs]: `FirstTimeEntryView` observable yields only `DistinctUntilChanged`

File: Toggl.Giskard/Activities/MainActivity.cs [hunk 2]
```diff
@@ -64,6 +71,24 @@ protected override void OnCreate(Bundle bundle)
             disposeBag.Add(ViewModel.WeakSubscribe<PropertyChangedEventArgs>(nameof(ViewModel.SyncingProgress), onSyncChanged));
 
             setupStartTimeEntryOnboardingStep();
+
+            mainRecyclerView = FindViewById<MainRecyclerView>(Resource.Id.MainRecyclerView);
+            mainRecyclerView.FirstTimeEntryView
```
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply Toggl.Giskard/Activities/MainActivity.cs]: How much could could a couldchuck chuck if a couldchuck could chuck could.

**Comment by pe0ny** [comment]
@kornelijepetak "Tap to edit" should always be in the new line. So all good 👍

File: Toggl.Giskard/Activities/MainActivity.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Giskard/Activities/MainActivity.cs]: :wat:

File: Toggl.Giskard/Activities/MainActivity.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (daividssilverio) [reply Toggl.Giskard/Activities/MainActivity.cs]: You are right, I didn't read it properly.

File: Toggl.Giskard/Activities/MainActivity.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply Toggl.Giskard/Activities/MainActivity.cs]: Read your comment 😛

File: Toggl.Giskard/Views/MainRecyclerView.cs [hunk 3]
```diff
@@ -53,6 +63,54 @@ public MainRecyclerView(Context context, IAttributeSet attrs, int defStyle)
...
+        private View findFirstTimeEntryView()
+        {
+            var layoutManager = (LinearLayoutManager)GetLayoutManager();
+
+            for (var position = 0; position <= MainRecyclerAdapter.ItemCount; position++)
+            {
+                var item = MainRecyclerAdapter.GetItem(position);
+                if (item is TimeEntryViewModel timeEntryViewModel)
+                {
+                    View view = layoutManager.FindViewByPosition(position);
```
Comment (daividssilverio) [inline Toggl.Giskard/Views/MainRecyclerView.cs]: `layoutManager.FindFirstVisibleItemPosition()` and `layoutManager.FindFirstVisibleItemPosition FindFirstCompletelyVisibleItemPosition` didn't workout for you? I guess you wouldn't need this loop if you used one of them.

File: Toggl.Giskard/Activities/MainActivity.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (kornelijepetak) [reply Toggl.Giskard/Activities/MainActivity.cs]: And you fix your comment and now all my comments sound stupid 😛

File: Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs [hunk 4]
```diff
@@ -17,8 +17,24 @@ public EditTimeEntryOnboardingStep(IOnboardingStorage onboardingStorage, IObserv
...
+
+        private bool shouldBeVisible(
+            bool signedUpUsingTheApp,
+            bool hasTappedTimeEntry,
+            bool navigatedAwayFromMainView,
+            bool hasTimeEntryBeenContinued,
+            bool isEmpty)
+        {
+            return signedUpUsingTheApp 
+                    && !navigatedAwayFromMainView
```
Comment (daividssilverio) [inline Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs]: 🤔It seems this logic is wrong. The user does have navigate away from the main view to start his first TE, so I think this condition is only valid after the user has seen the tooltip.

Comment (daividssilverio) [reply Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs]: One possible way to go would be:
1. Rename `navigatedAwayFromMainView` to `navigatedAwayFrom...`
2. Create another onboarding storage variable, like `hasSeenEditTimeEntryOnboardingStep`, or similar, that would be set on the first time the tooltip is shown.
3. Then, the only set `navigatedAwayFromMainView` to true when `hasSeenEditTimeEntryOnboardingStep` is also true.

**Comment by daividssilverio** [comment]
I tested on the emulator with a new user, signing up from the app.

File: Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs [hunk 5]
```diff
@@ -17,8 +17,24 @@ public EditTimeEntryOnboardingStep(IOnboardingStorage onboardingStorage, IObserv
...
+                onboardingStorage.HasTimeEntryBeenContinued,
                 isTimeEntriesLogEmpty,
-                (signedUpUsingTheApp, hasTappedTimeEntry, isEmpty) => signedUpUsingTheApp && !hasTappedTimeEntry && !isEmpty);
+                shouldBeVisible);
+        }
+
+        private bool shouldBeVisible(
+            bool signedUpUsingTheApp,
+            bool hasTappedTimeEntry,
+            bool navigatedAwayFromMainView,
```
Comment (juanlaube) [inline Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs]: This is true after navigating to the start TE activity, thus making the tooltip not visible.

Comment (kornelijepetak) [reply Toggl.Foundation.MvvmCross/Onboarding/MainView/EditTimeEntryOnboardingStep.cs]: This is actually the same issue @daividssilverio reported. I will fix this ASAP.

**Pull request merged**